src/stdlib_io.fypp

```diff
@@ -145,11 +146,35 @@ contains
 
       do i = 1, max_rows_
         #:if 'real' in t1
-          read(s, "(*"//FMT_REAL_${k1}$(1:len(FMT_REAL_${k1}$)-1)//",1x))") d(i, :)
+        if ( present( fmt ) ) then
+          if ( fmt == '*' ) then
+            read (s,*) d(i, :)
+          else
+            read (s,fmt) d(i, :)
+          endif
+        else 
+          read (s,"(*"//FMT_REAL_${k1}$(1:len(FMT_REAL_${k1}$)-1)//",1x))") d(i, :)
+        end if
```

Comment:
Checking the same `if` statement at every iteration would be quite slow. So I suggest to move this if statement outside the loop.
Could something like this work:
```fortran
fmt_ = optval('*('//fmt//',1x)', '*', fmt)

do i = 1, max_rows_
  read (s, fmt_) d(i,:)
enddo
```